flag_groups.go

```diff
@@ -158,13 +161,17 @@ func validateRequiredFlagGroups(data map[string]map[string]bool) error {
 
 		// Sort values, so they can be tested/scripted against consistently.
 		sort.Strings(unset)
-		return fmt.Errorf("if any flags in the group [%v] are set they must all be set; missing %v", flagList, unset)
+		return &FlagGroupError{
+			flagList:      flagList,
+			flagGroupType: FlagsAreRequiredTogether,
+			problemFlags:  unset,
```

Comment:
Here and everywhere, I don't see a logic to use a pointer

```suggestion
		return FlagGroupError{
			flagList:      flagList,
			flagGroupType: FlagsAreRequiredTogether,
			problemFlags:  unset,
```

It leads to complicated syntax later, and double pointers so you have to do this

```go
  var invalidArgCountErr *cobra.InvalidArgCountError 
   if errors.As(err, &invalidArgCountErr) {
```

When this could be enough

```go
  var invalidArgCountErr cobra.InvalidArgCountError 
   if errors.As(err, &invalidArgCountErr) {
```

Reply:
I totally agree that it makes the syntax complicated and I'm perfectly happy to change it, but let me explain why I did it this way first.

The main reason I chose to use a pointer receiver for the error structs is because that's what the Go standard library does it for its error types, and I wanted to keep Cobra consistent with that. A few examples of the top of my head are [os/PathError](https://pkg.go.dev/os#PathError) (which is used in the example for [errors.As](https://pkg.go.dev/errors#As)), [json/SyntaxError](https://pkg.go.dev/encoding/json#SyntaxError.Error), and [exec/ExecError](https://pkg.go.dev/os/exec#ExitError).

I haven't been able to find any official documentation explaining why the standard library chooses to use pointer receivers for errors, but [this Reddit comment](https://www.reddit.com/r/golang/comments/16eoxsv/comment/jzx628d/) explains a compile-time type safety benefit of doing it that way.

If you still want me to change it to use a value receiver though, just let me know.

Reply:
I know a lot of code in Go standard library are not respecting the Go idiomatic way. So sometimes finding examples in Go code doesn't help much.

Droping @alexandear here.

What do you think about this?

Reply:
@eth-p you're completely right about `os.PathError` and others. But I'm sticking to @ccoVeille's approach of returning a non-pointer `FlagGroupError` as a simpler option.

Other suggestions:

- Maybe we should make the `FlagGroupError` struct unexported (i.e., `flagGroupError`)? In the future, we can refactor to return a pointer if needed without breaking compatibility.
- Should we return `error` instead of `*FlagGroupError`?